packages/libs/core/src/match.ts

```diff
@@ -32,7 +32,7 @@ export function compileDestination(
     ) {
       // Handle external URL redirects
       const { origin, pathname, search } = new URL(destination);
-      const toPath = compile(pathname, { encode: encodeURIComponent });
+      const toPath = compile(pathname);
```

Comment:
**Question:** Instead of removing the encoding entirely, would it make sense to use `encodeURI`? 

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent

Reply:
Hi, thanks for the suggestion.

I've checked how the next.js server works with path matching and the difference is quite significant.
However, the usage of `encodeURI` makes the behavior a little bit closer.
It will encode not-ASCII chars but will not encode symbols like @=.